cmake/modules/extensions.cmake

```diff
@@ -3335,6 +3335,46 @@ macro(zephyr_linker_memory_ifdef feature_toggle)
   endif()
 endmacro()
 
+# Usage:
+#   zephyr_linker_dts_section(PATH <path>)
+#
+# Zephyr linker devicetree memory section from path.
+#
+# This function specifies an output section for the platform in use based on its
+# devicetree configuration.
+#
+# The section will only be defined if the devicetree exists and has status okay.
+#
+# PATH <path>      : Devicetree node path.
+#
+function(zephyr_linker_dts_section)
+  set(single_args "PATH")
+  cmake_parse_arguments(DTS_SECTION "" "${single_args}" "" ${ARGN})
+
+  if(DTS_SECTION_UNPARSED_ARGUMENTS)
+    message(FATAL_ERROR "zephyr_linker_dts_section(${ARGV0} ...) given unknown "
+	    "arguments: ${DTS_SECTION_UNPARSED_ARGUMENTS}"
+    )
+  endif()
+
+  if(NOT DEFINED DTS_SECTION_PATH)
+    return()
+  endif()
+
+  dt_node_exists(exists PATH ${DTS_SECTION_PATH})
+  if(NOT ${exists})
+    return()
+  endif()
+
+  dt_reg_addr(addr PATH ${DTS_SECTION_PATH})
+
+  dt_prop(name PATH ${DTS_SECTION_PATH} PROPERTY "zephyr,memory-region")
+  zephyr_string(SANITIZE name ${name})
```

Comment:
Still an issue with this code.
It is not verified that the given node has this property, and if it doesn't it could be because the node doesn't exists, or doesn't have the property.

I don't believe we should allow this function to be called for node that doesn't have the `zephyr,memory-region` property.
See for example this:
https://github.com/zephyrproject-rtos/zephyr/blob/5e79463ace625b649028e599e5ad75e52d3bc168/cmake/modules/extensions.cmake#L2732-L2741